### PR TITLE
Refactor minimum recording length handling

### PR DESCRIPTION
**Pull request by NeonDaniel**
Move timer reset to ensure minimum recording length is measured from the moment input is accepted

Related to #107
Related to #110
Also reported by a Neon user [on Matrix](https://matrix.to/#/!ZhEZYNzKBfpEAhtQIz:matrix.org/$MdRLj3Tv6zDfODvrVAuPapf_ILzPfC6yz70jkvWZJKQ?via=matrix.org&via=nitro.chat&via=rx.haunted.computer) that no input was accepted after WW confirmation sound was played

### CODE REVIEW

**Comment by NeonDaniel** [comment]
#76 and #73 added these Timers and tests, but I don't see why we wouldn't wait for the confirmation or playback before proceeding with recording which is what appears to happen now.

@JarbasAl @emphasize am I missing something with the Timer implementation?

**Review by mikejgray — COMMENTED** [review]
Code looks fine, will test when I can

File: ovos_dinkum_listener/service.py [hunk 1]
```diff
@@ -791,11 +794,14 @@ def _handle_listen(self, message: Message):
...
                     self.voice_loop.confirmation_event.clear()
-                    Timer(0.5, lambda: self.voice_loop.confirmation_event.set()).start()
+                    if not self.voice_loop.confirmation_event.wait(0.5):
+                        LOG.warning("Confirmation event timed out!")
+                        self.voice_loop.confirmation_event.set()
                 else:
                     self.voice_loop.state = ListeningState.BEFORE_COMMAND
         else:
             self.voice_loop.state = ListeningState.BEFORE_COMMAND
+        self.voice_loop.reset_speech_timer()
```
Comment (JarbasAl) [inline ovos_dinkum_listener/service.py:804]: why was this moved? what are the implications of moving it?

File: ovos_dinkum_listener/service.py [hunk 2]
```diff
@@ -791,11 +794,14 @@ def _handle_listen(self, message: Message):
                 if not self.config["listener"].get("instant_listen"):
                     self.voice_loop.state = ListeningState.CONFIRMATION
                     self.voice_loop.confirmation_event.clear()
-                    Timer(0.5, lambda: self.voice_loop.confirmation_event.set()).start()
+                    if not self.voice_loop.confirmation_event.wait(0.5):
+                        LOG.warning("Confirmation event timed out!")
+                        self.voice_loop.confirmation_event.set()
```
Comment (JarbasAl) [inline ovos_dinkum_listener/service.py:799]: if this is added here, then doesnt make sense here https://github.com/OpenVoiceOS/ovos-dinkum-listener/blob/ad0954dff1da596eca9975bfb12960fe3e8eb8dd/ovos_dinkum_listener/voice_loop/voice_loop.py#L250

i think the idea is to allow the loop to continue uninterrupted instead of blocking for 0.5 seconds, as this event is supposed to come from an external source (ovos-audio)

i would suggest making the timeout configurable
 
```python
if self.chunk_callback is not None:
    self._chunk_info.energy = self.debiased_energy(chunk, self.mic.sample_width)
    self.chunk_callback(self._chunk_info)
```

 
 @emphasize should be more familiar with this part of the code and give a better rationale on why it was done this way

Comment (JarbasAl) [reply ovos_dinkum_listener/service.py:799]: this affects usage of get_response via voice satellites, that was the main reason for introducing this new state in the loop, as get_response would often timeout just because the sat device started listening too late, 

if the `instant_listening` flag is not set then it's supposed to ensure audio is only recorded once the acknowledgment sound is played

**Comment by JarbasAl** [comment]
replaced by #125

File: ovos_dinkum_listener/service.py (hunk 1 above)
Comment (NeonDaniel) [reply ovos_dinkum_listener/service.py:804]: This was moved so the timer is reset after the audio playback is completed, so the time it takes to play the confirmation sound isn't factored into the minimum length
